sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -104,6 +104,15 @@ trait CheckAnalysis extends PredicateHelper {
       case u: UnresolvedTable =>
         u.failAnalysis(s"Table not found for '${u.commandName}': ${u.multipartIdentifier.quoted}")
 
+      case u @ UnresolvedView(NonSessionCatalogAndIdentifier(catalog, ident), cmd, _) =>
+        u.failAnalysis(
+          s"Cannot specify catalog `${catalog.name}` for view ${ident.quoted} " +
+            "because view support in v2 catalog has not been implemented yet. " +
+            s"$cmd expects a view.")
+
+      case u: UnresolvedView =>
```

Comment:
do we have a clear rule about when `hint` should be used in error message?

Reply:
The rule I am thinking is if an expected type of relation is not found. For example, `UnresolvedTable` is resolved to `ResolvedView` or `UnresolvedView` is resolved to `ResolvedTable`.

Reply:
then probably we should make the name clearer: `relationTypeMismatchHint`

Reply:
That sounds good, thanks!